MODIFIED convert-json-data-to-html to warn about XSS attacks (english) (#37036)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/04-data-visualization/json-apis-and-ajax/convert-json-data-to-html.english.md

  html += "</div><br>";
});
```

> <b>WARNING!</b> To complete this challenge you need to add new HTML elements. To do this you will use the innerHTML method, this can make your site vulnerable to XSS attacks, which in brief allows an attacker to execute code on the client side. To learn more <a href="https://guide.freecodecamp.org/security/xss-cross-site-scripting/">see the guide</a>

Reviewer: I suggest using the verbiage of last suggestion made in the issue by @ojeytonwilliams .  It is much clearer.  Also, since the Guide will no longer be in the next release, the link to the Guide should be removed.  We prefer to use `Note:` instead of `Warning!` to be consistent with other challenges.
```suggestion
<b>Note:</b> For this challenge you need to add new html elements to the page, so you cannot rely on textContent. Instead you need to use innerHTML which can make a site vulnerable to cross site scripting attacks.
```

Author: Will update the spanish version to match

});
```

> <b>Note:</b> For this challenge you need to add new html elements to the page, so you cannot rely on textContent. Instead you need to use innerHTML which can make a site vulnerable to cross site scripting attacks.

Reviewer: Something I did not catch in the first review, but according to our guidelines, notes are surrounded by strong tags.
```suggestion
<strong>Note:</strong> For this challenge you need to add new html elements to the page, so you cannot rely on textContent. Instead you need to use innerHTML which can make a site vulnerable to cross site scripting attacks.
```

> <strong>Note:</strong> For this challenge you need to add new html elements to the page, so you cannot rely on textContent. Instead you need to use innerHTML which can make a site vulnerable to cross site scripting attacks.

Reviewer 2: I think it would be better to capitalize HTML and add backticks (`) around `textContent` and `innerHTML`. This will render the two methods as inline code elements and will make it more consistent with other parts of the curriculum:

```
<strong>Note:</strong> For this challenge you need to add new HTML elements to the page, so you cannot rely on `textContent`. Instead you need to use `innerHTML` which can make a site vulnerable to cross site scripting attacks.
```

Reviewer: @Wocanilo Once you make the change suggested above on this PR, you will need to change the translation on PR https://github.com/freeCodeCamp/freeCodeCamp/pull/37037 to match.

> <strong>Note:</strong> For this challenge you need to add new HTML elements to the page, so you cannot rely on `textContent`. Instead you need to use `innerHTML`, which can make a site vulnerable to cross site scripting attacks.

Reviewer 3: ```suggestion
<strong>Note:</strong> For this challenge, you need to add new HTML elements to the page, so you cannot rely on `textContent`. Instead you need to use `innerHTML`, which can make a site vulnerable to Cross-site scripting attacks.
```

Reviewer: @Manish-Giri Can you also add a comma after the word `Instead` in the second sentence?

Reviewer 4: This is not my PR, and I cannot commit this suggestion via the interface. I made the original suggested changes on behalf of the OP in hopes to get this PR merged and I think it's fine how it is. If you want it this way, you should make the changes.

Reviewer: The suggestions @Manish-Giri made will improve the challenge.

Reviewer: @Manish-Giri I have added a new commit with your suggested changes.  If you are OK with the PR now, can you approve and merge it?  Thanks.
